hooks/succ-pre-tool.cjs

```
@@ -206,6 +269,172 @@ const DATA_PREFIXES = [
+  { re: /<\|system\|>/i, desc: 'System role token' },
+  {
+    re: /<\/(?:hook-rule|file-context|soul|previous-session|session|compact-fallback|security-warning|commit-format|pre-commit-review|succ-agents)>/i,
+    desc: 'Closing succ XML tag',
+  },
+  { re: /<\/?system>/i, desc: 'XML system tag' },
+  { re: /<\/?assistant>/i, desc: 'XML assistant tag' },
+  { re: /<\/?user>/i, desc: 'XML user tag' },
+  { re: /\[system\]/i, desc: 'Bracketed system role' },
+  { re: /\[assistant\]/i, desc: 'Bracketed assistant role' },
+  // Hidden content / obfuscation
+  { re: /[\u200B\u200C\u200D\uFEFF\u00AD\u2060]{3,}/, desc: 'Zero-width char cluster' },
```

Biome (2.4.4)

[error] A character class cannot match a joined character sequence.

(lint/suspicious/noMisleadingCharacterClass)